test/test_general.cpp

```diff
@@ -52,3 +53,65 @@ TEST(General, TestLineParsing) {
   EXPECT_EQ(data[1], "Entry2");
   EXPECT_EQ(data[2], "EscapedEntry1;EscapedEntry2");
 }
+
+TEST(General, Dijkstra) {
+  /*
+            (6)
+        .----------.
+       \/          \/
+  0 <-> 1 --> 2 <-> 3
+    (1)   (2)   (3)
+  */
+  CouplingMap cm = {{0, 1}, {1, 0}, {1, 2}, {2, 3}, {3, 2}, {1, 3}, {3, 1}};
+  Matrix      edgeWeights = {
+      {0, 1, 0, 0}, {1, 0, 2, 6}, {0, 15, 0, 3}, {0, 6, 3, 0}};
```

Comment:
Is the `15` in the third entry here on purpose? Maybe I just interpreted the comment above in the wrong way, but to me it matches every entry but this `15`.

Reply:
Good catch. Thank you for the hint. But this is in fact on purpose. Our Dijkstra implementation generally ignores directionality of edges. The edge weight of `15` (which is higher than the sum of all other weights combined) kind of simulates a unidirectional edge. (Just to make sure that Dijkstra takes the correct path in that edge case)

Reply:
That makes sense. Thanks for the explanation!